spring-integration-core/src/main/java/org/springframework/integration/support/management/graph/IntegrationGraphServer.java

```diff
@@ -162,6 +164,19 @@ private void gateways(Collection<IntegrationNode> nodes, Collection<LinkNode> li
 			nodes.add(gatewayNode);
 			producerLink(links, channelNodes, gatewayNode);
 		}
+		Map<String, GatewayProxyFactoryBean> gpfbs = this.applicationContext
+				.getBeansOfType(GatewayProxyFactoryBean.class);
+		for (Entry<String, GatewayProxyFactoryBean> entry : gpfbs.entrySet()) {
+			Map<Method, MessagingGatewaySupport> methodMap = entry.getValue().getGateways();
+			int i = 0;
+			for (Entry<Method, MessagingGatewaySupport> gwEntry : methodMap.entrySet()) {
+				MessagingGatewaySupport gateway = gwEntry.getValue();
+				MessageGatewayNode gatewayNode = this.nodeFactory.gatewayNode(
+						entry.getKey().substring(1) + "." + gwEntry.getKey().getName() + "#" + i++, gateway);
```

Comment:
>M-m-m. Not sure in the i++ for each method:

@artembilan I didn't think it was worth the effort to only disambiguate methods with the same name.

But I do think using the full signature would be better - especially because the methods are likely returned in different orders from one execution to the next.

Since it looks like you already wrote the code, feel free to push with your polish.

>How about a bit different type for this component,

I don't think that's really necessary - we don't differentiate between, say, inbound Http, JMS, RabbitMQ, etc gateways. And, they are all `MessagingGatewaySupport` objects.

The user can always use appropriate bean names.